docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,692 @@
+# Week 7: Database Basics and Connecting the Backend to the Database
+
+## Agenda
+- Database basics
+  - How to think in terms of dbs
+  - How to think about tables
+  - Data types
+  - How to create a table/db
+  - Basic query structure
+- Backend + db connection
+  - JDBC and making calls to the db
+  - Prepared statements
+  
+## Basic Plan
+1. Databases!
+  - Excel Sheets to Database Tables
+  - Data Types in SQL
+    - `SERIAL`
+    - `CHAR(x)`
+    - `VARCHAR(x)`
+      - `CHAR` vs `VARCHAR`
+    - `TEXT`
+      - `CHAR`/`VARCHAR` vs `TEXT`
+    - `INTEGER`/`BIGINTEGER`
+    - `DECIMAL`
+    - `TIMESTAMP`
+    - `BOOLEAN`/`TINYINTEGER`
+    - `ENUM`
+  - Column modifiers
+    - `PRIMARY KEY`
+    - `FOREIGN KEY`
+    - `NOT NULL`
+    - `AUTO INCREMENT` (`SERIAL`)
+    - `DEFAULT`
+  - Creating a Database and Table
+  - Query structure
+    - The `INSERT` statement
+    - The `SELECT` statement
+    - The `UPDATE` statement
+    - The `DELETE` statement
+2. Backend + DB Connection
+  - JDBC and Working with the DB in Java
+    - Properties and Environment Files
+    - `Connection`s and the `DriverManager`
+  - Prepared Statements
+  
+## Databases! 
+
+Databases are persistent storage. That means that they continue to hold application data on disk, even if your app stops
+running (whether intentionally or unintentionally). They have APIs, mostly through special software Open DataBase 
+Connectivity (ODBC) and Java DataBase Connectivity (JDBC, what we'll be using). The type of database we're going to be 
+working with is called a relational database, which means it acts like a collection of Excel sheets. It also has a nice
+little syntax for performing operations called SQL (pronounced "see-quel" or "ess-que-ell"). There are also 
+other types of DBs, called NoSQL databases, which act a little differently, but we'll just stick with relational in this
+program.
+
+### Excel Sheets to Database Tables
+
+When you think of Excel, you think of a bunch of rows/entries and columns/fields which make up a sheet. In terms of DBs, 
+each table is like a separate sheet containing pretty uniform data. Each column (the up/down ones) acts like a separate 
+field. You specify a datatype and name for your columns, and then as data gets entered, rows (the long left/right) ones 
+get created, read, updated, and deleted (remember CRUD). One thing to note is that in relational databases, unlike 
+Excel, you don't have an unlimited amount of columns that can be automatically added. When you create your table, it has 
+a set structure, meaning you can only add/remove/update columns by changing the structure of the table. You can, 
```

Comment:
This sentence might be a little long. I think maybe saying "A table has a set structure once you create it. This means..."